src/iam/resources.py

```diff
@@ -252,3 +254,24 @@ def get(self):
             return User.query.filter(User.id == g.jwt_claims['usr']).one(), 200
         except NoResultFound:
             return f"No user with id {g.jwt_claims['usr']}", 404
+
+
+@doc(description="Register a user and authenticate in the local database")
+class UserRegisterResource(MethodResource):
+    @use_kwargs(UserRegisterSchema)
+    def post(self, first_name, last_name, email, password):
+        # Check if specified email already exists
+        user = User.query.filter_by(email=email).first()
+        if user:
+            return f"User with provided email already exists", 400
+
+        user = User(
+            first_name=first_name,
+            last_name=last_name,
+            email=email,
+            password=hasher.encode(password)
```

Comment:
You should use `user.set_password(password)` instead

Reply:
True, thanks, fixed